### PR TITLE
HPCC-26790 Provide HPCC Metrics via Azure Insights write-up

### PR DESCRIPTION
**Pull request by rpastrana**
- Updates structure of metrics README
- Provides detailed steps to route HPCC metrics to Azure insights

Signed-off-by: Rodrigo Pastrana <Rodrigo.Pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@JamesDeFabia / @g-pan please review

**Comment by rpastrana** [comment]
@jakesmith or @ghalliday please review

File: helm/examples/metrics/README.md [hunk 1]
```diff
@@ -55,5 +112,68 @@ For example, the following Prometheus scrape job can be applied to Prometheus de
...
+### Azure Insights (Monitoring and Analytics)
+HPCC metrics can be easily routed to Azure Insights for application level monitoring and analytics via the aforementioned Prometheus metrics services. This process requires the HPCC Prometheus metrics collection system, but does not involve any instances the Prometheus server. Since Azure Insights is able to scrape metrics from Prometheus styled services, this process relies on enabling Azure Insights on the target AKS cluster, and configuring the automatic scraping of metrics from the Prometheus metric endpoints provided by HPCC components.
+
+- Enable HPCC's Prometheus metrics on the target HPCC deployment.
+  - For example (utilizing prometheus_metrics.yml values file provided in hpcc-systems/HPCC-Platform github repo:
+    ```code
+    helm install mycluster ./hpcc -f helm/examples/metrics/prometheus_metrics.yml
+    ```
+- Enable Azure's Insights on the target AKS cluster
+  - Can be done from Azure portal, or via CLI
```
Comment (JamesDeFabia) [inline helm/examples/metrics/README.md:127]: no need for a comma.

File: helm/examples/metrics/README.md [hunk 2]
```diff
@@ -22,15 +22,72 @@ helm install mycluster ./hpcc -f <path>/prometheus_metrics.yml
...
+      esp_requests_active{containername="eclqueries", job="hpcc-per-pod-metrics"}	3
+      esp_requests_active{containername="eclservices",  job="hpcc-per-pod-metrics"}	0
+      esp_requests_active{containername="eclwatch", job="hpcc-per-pod-metrics"}	3
+      esp_requests_active{containername="esdl-sandbox",  job="hpcc-per-pod-metrics"}	3
+      esp_requests_active{containername="sql2ecl", job="hpcc-per-pod-metrics"} 0
+      ```
+- The available Grafana service provides advanced tools for higher level metrics processing and alerting
+- please see [Managed Prometheus Helm chart](https://github.com/hpcc-systems/HPCC-Platform/tree/master/helm/managed/metrics/prometheus) for further details.
+
+### Self Managed Prometheus
```
Comment (JamesDeFabia) [inline helm/examples/metrics/README.md:88]: s/b Self-managed

File: helm/examples/metrics/README.md [hunk 3]
```diff
@@ -22,15 +22,72 @@ helm install mycluster ./hpcc -f <path>/prometheus_metrics.yml
...
+One of the advantages of the aforementioned managed Prometheus Helm chart, is the pre-configured scrape job 'hpcc-per-pod-metrics' which instructs Prometheus to scrape metrics from all HPCC pods which report metrics.
+On a self managed Prometheus set-up, an "additionalScrapeConfigs" entry is necessary to link the HPCC metrics services to Prometheus
```
Comment (JamesDeFabia) [inline helm/examples/metrics/README.md:90]: s/b self-managed

**Review by JamesDeFabia — CHANGES_REQUESTED** [review]

**Comment by JamesDeFabia** [comment]
Also, at the beginning, it might be better to say:
...collection system such as Prometheus, Elasticsearch, or Azure Insights.

**Review by JamesDeFabia — APPROVED** [review]
Looks good from my POV

File: helm/examples/metrics/container-azm-ms-agentconfig-prometheusperpod.yaml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      tcp_listener_buffer_size = 10
+      tcp_listener_mem_buf_limit = 200
+
+    # The following settings are "undocumented", we don't recommend uncommenting them unless directed by Microsoft.
+    # They increase the maximum stdout/stderr log collection rate but will also cause higher cpu/memory usage.
+    # [agent_settings.fbit_config]
+    #   log_flush_interval_secs = "1"                 # default value is 15
+    #   tail_mem_buf_limit_megabytes = "10"           # default value is 10
+    #   tail_buf_chunksize_megabytes = "1"            # default value is 32kb (comment out this line for default)
+    #   tail_buf_maxsize_megabytes = "1"              # defautl value is 32kb (comment out this line for default)
```
Comment (ghalliday) [inline helm/examples/metrics/container-azm-ms-agentconfig-prometheusperpod.yaml:154]: typo: default

File: helm/examples/metrics/container-azm-ms-agentconfig-prometheusperpod.yaml [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        container_memory_working_set_threshold_percentage = 95.0
+
+    # Alertable metrics configuration settings for persistent volume utilization
+    [alertable_metrics_configuration_settings.pv_utilization_thresholds]
+        # Threshold for persistent volume usage bytes, metric will be sent only when persistent volume utilization exceeds or becomes equal to the following percentage
+        pv_usage_threshold_percentage = 60.0
+
+    # Alertable metrics configuration settings for completed jobs count
+    [alertable_metrics_configuration_settings.job_completion_threshold]
+        # Threshold for completed job count , metric will be sent only for those jobs which were completed earlier than the following threshold
```
Comment (ghalliday) [inline helm/examples/metrics/container-azm-ms-agentconfig-prometheusperpod.yaml:132]: trivia, space before comma.

**Review by ghalliday — APPROVED** [review]
@rpastrana looks good.  A couple of trivial comments to prove I read it!

**Comment by rpastrana** [comment]
Resolved minor issues and squashed.
@richardkchapman ready for merge

**Pull request merged**